### DISCUSSION TITLE
Faker date returns wrong format

### DISCUSSION THREAD

**Discussion by csophycgi** [mockoon/mockoon#396] [Help] [answered]
Hello everyone,
First of all, thank you for this tool ! I used it for my new project and its changing my life 😃  

There is one thing about the date by faker that seems to be wrong, when using it like this :
{{faker 'date.soon'}} 
I get this result : "Thu Jan 07 2021 15: 31: 15 GMT+0100 (Central European Standard Time)"
I was expecting an ISO format like : "2020-07-08T16:31:54.899Z"

Am i doing something wrong or is there another way to get the ISO format ?

Thank you

Chris

**Comment by 255kb** [comment, marked as answer]
Hi @csophycgi ! We are glad Mockoon changed your life 💪 
Concerning the Faker helpers, I think it's the expected date format. There is currently no formatting option for these helpers and unfortunately, Faker is not maintained anymore since recently. 
However, there is an alternative helper in Mockoon: `{{date '2020-11-20' '2020-11-25' 'dd-MM-yyyy'}}`
It supports formatting, and the two first parameters are `from` `to`. 

If this does not solve your use case, feel free to open an issue.

**Comment by csophycgi** [comment]
Indeed from a their docs it is the expected format but weirdly in their demo app its different : 
http://marak.com/faker.js/#date

![image](https://user-images.githubusercontent.com/32295314/104021772-a3953480-51bf-11eb-9263-2a6e478a161f.png)

I tried the alternative option without success because DummyJson formatting is not working properly to achieve what I want.

**Reply by 255kb** [reply]
Indeed, I never ever see this demo page, it was down for a long time. But looking at both the [code](https://github.com/Marak/faker.js/blob/master/lib/date.js#L80) in the repository and the one in the js files loaded in the demo, there is no formatting done. It's a basic `new Date()`. 

Concerning the `date` helper, it existed when we used DummyJSON, but I reimplemented the helpers to keep the app backward compatible while getting rid of the dependency. 
Now the formatting uses [date-fns](https://date-fns.org/docs/format) formatting options, doing this should work:
`{{ date '2020' '2021' 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'' }}`
(You have to use single quotes for T and Z, and may need to espace them depending on where you are calling the helper.

**Reply by csophycgi** [reply]
oh thanks, it works !
faker date.past/future is cool but this alternative will do.

Thank you
